arpeggio/__init__.py

```diff
@@ -403,7 +405,37 @@ def _parse(self, parser):
             parser._nm_raise(self, c_pos, parser)
 
         return result
+        
+class ShortestChoice(Sequence):
+    def _parse(self, parser):
+        match = False
+        c_pos = parser.position
+        shortest_result = None
+        shortest_result_str = None
+        shortest_result_cpos = None
+        for e in self.nodes:
+            try:
+                result = e.parse(parser)
+                if result is not None:
+                    result = [result]
+                    result_str = "".join([x.flat_str() for x in flatten(result)])
+                    if (not shortest_result) or (len(shortest_result_str) > len(result_str)):
```

Comment:
Please make a unit test also if you can. I am trying to keep test coverage as high as possible. Thanks.